core/LSU.cpp

```diff
@@ -258,17 +259,18 @@ namespace olympia
     {
         sparta_assert(inst_ptr->getStatus() == Inst::Status::RETIRED,
                       "Get ROB Ack, but the store inst hasn't retired yet!");
+        if(!inst_ptr->isVector()){
```

Comment:
You should assert if the inst is a vector since you have separate ports for scalar and vector.